pkg/destroy/aws/aws.go

```diff
@@ -1827,6 +1827,42 @@ func deleteS3(session *session.Session, arn arn.ARN, logger logrus.FieldLogger)
 	}
 	logger.Debug("Emptied")
 
+	var lastError error
+	err = client.ListObjectVersionsPagesWithContext(aws.BackgroundContext(), &s3.ListObjectVersionsInput{
+		Bucket: aws.String(arn.Resource)}, func(page *s3.ListObjectVersionsOutput, lastPage bool) bool {
+		var deleteObjects []*s3.ObjectIdentifier
+		for _, deleteMarker := range page.DeleteMarkers {
+			deleteObjects = append(deleteObjects, &s3.ObjectIdentifier{
+				Key:       aws.String(*deleteMarker.Key),
+				VersionId: aws.String(*deleteMarker.VersionId),
+			})
+		}
+		for _, version := range page.Versions {
+			deleteObjects = append(deleteObjects, &s3.ObjectIdentifier{
+				Key:       aws.String(*version.Key),
+				VersionId: aws.String(*version.VersionId),
+			})
+		}
+		_, err := client.DeleteObjects(&s3.DeleteObjectsInput{
+			Bucket: aws.String(arn.Resource),
+			Delete: &s3.Delete{
+				Objects: deleteObjects,
+				Quiet:   aws.Bool(true),
+			},
+		})
+		if err != nil {
+			lastError = errors.Wrapf(err, "delete object failed %v", err)
+		}
+		return !lastPage
+	})
+	if lastError != nil {
+		return lastError
+	}
+	if err != nil {
+		return err
+	}
```

Comment:
this should not error when we get bucket no found error like https://github.com/openshift/installer/pull/3393/files#diff-ab65f0e2ba0237c5cc1429673c22cc08R1825 (line 1825 right above)

Reply:
Fixed!